src/v/archival/CMakeLists.txt

```diff
@@ -1,27 +1 @@
-
-v_cc_library(
-  NAME archival
```

Comment:
Could you elaborate on why this library is folded into cluster? I'm not sure why this is needed (and what deep dependencies is implying from the PR description).

Reply:
archival/ and cluster/ started off as separate sub-systems and separate libraries. but over time they started have dependencies on each other. for the most part we've been able to fix these circular dependencies, but the ones between archival and cluster are much deeper. to the point that it seems that archival may more natrually fit into the cluster subsystem.

Reply:
Thanks for the context!